File: Lab07/count_word.pl
#!/usr/bin/perl -w

if(@ARGV != 1) {
   die "Usage: $0 <word>\n";
}

$wordCount = 0;

$searchWord = lc $ARGV[0];

while($line = <STDIN>) {

   foreach my $word ($line =~ /[a-z]+/ig) {  #capture all occurences, case insensitive
      $word = lc $word;
      $wordCount++ if $word eq $searchWord;
   }
}

print "$searchWord occurred $wordCount times\n";

File: Lab07/count_word.py
#!/usr/bin/python

import sys,re

if len(sys.argv) != 2:
   print "Usage %s <word>" %sys.argv[0]
   sys.exit(1)
   
searchWord = sys.argv[1].lower()
wordCount = 0

for line in sys.stdin:
   for word in re.findall(r'[a-zA-Z]+',line):   
      word = word.lower()
      if word == searchWord:
         wordCount += 1

print searchWord, "occurred", wordCount, "times"
